### DISCUSSION TITLE
Request: Re-Record Episode 100 to 103 and Update Source Code

### DISCUSSION THREAD

**Discussion by vermont42** [pointfreeco/episode-code-samples#120] [Ideas]
My company is considering a SwiftUI rewrite. In light of the iOS-developer community's embrace of TCA and my own high regard for Point-Free, I am considering recommending TCA as our architecture.

I decided to become more familiar with TCA by working through episodes 100 to 103, as suggested in TCA's readme. Not surprisingly, given the fact that the episodes were recorded two years ago, there are many deprecation warnings. More concerningly, one of the unit tests fails, not just in the video's code but also in the code on GitHub.

I recognize that re-recording every episode of Point-Free affected by a change to TCA, Swift, and SwiftUI would be impractical. But episodes 100 to 103 are a _recommended_ introduction in TCA's readme as it exists today. With those episodes in their current state, I am hesitant to point my coworkers to them. But learning about TCA  is a prerequisite to our company's adoption of TCA.

Given the importance of these four episodes, would Point-Free consider re-recording them without the deprecation warnings/failing unit test and updating the source code on GitHub?

**Comment by stephencelis** [comment]
@vermont42 Thanks for the feedback! We have a few big changes we want to make to TCA this year that are big enough to warrant new introductory episodes, so we do plan for such a thing, but it may be a few months before we're ready to.

Episode code samples are "time capsules" in a way, and it is difficult to ensure each code sample will build indefinitely, though it looks like we could take some time to lock SPM dependencies of certain episodes to make breakage less likely.

One thing to note in the case of these episodes is that the "Todos" app we build is also an "example" that comes with the main TCA repo. The examples in the TCA repo are kept up-to-date with the library, so you and your team can always consult this "Todos" app for changes that have been made to the library since those early episodes.

> More concerningly, one of the unit tests fails, not just in the video's code but also in the code on GitHub.

That test appears to be failing because it's using an `assert` API that was deprecated last December. Since then a community member contributed a feature to the new `TestStore` APIs that improves exhaustiveness of testing, which is getting erroneously triggered here. It looks like we could minimally backport support to the deprecated APIs, though, so I'll PR a fix. We should also lock the episode code sample to a much earlier version of TCA, which would have allowed the test to remain passing.

I'm actually somewhat surprised that code written for 0.1.0 still compiles in the latest version :smile: We do our best to keep deprecations around for a long time and try to maintain as much backwards-compatibility as possible.

**Comment by stephencelis** [comment]
I've pushed a fix here: https://github.com/pointfreeco/swift-composable-architecture/pull/1153

And verified that tests pass for that code sample when pointed to this branch. Thanks for the heads-up!

We'll discuss a path forward for general episode code sample SPM resolution soon.

**Comment by stephencelis** [comment]
Going to move this to be a discussion, since it's not a bug.
